buffer/src/main/java/io/netty/buffer/PoolChunk.java

```diff
@@ -231,18 +231,18 @@ final class PoolChunk<T> implements PoolChunkMetric {
         cachedNioBuffers = null;
     }
 
-    private static LongPriorityQueue[] newRunsAvailqueueArray(int size) {
-        LongPriorityQueue[] queueArray = new LongPriorityQueue[size];
+    private static IntPriorityQueue[] newRunsAvailqueueArray(int size) {
+        IntPriorityQueue[] queueArray = new IntPriorityQueue[size];
         for (int i = 0; i < queueArray.length; i++) {
-            queueArray[i] = new LongPriorityQueue();
+            queueArray[i] = new IntPriorityQueue();
         }
         return queueArray;
     }
 
     private void insertAvailRun(int runOffset, int pages, long handle) {
         int pageIdxFloor = arena.pages2pageIdxFloor(pages);
-        LongPriorityQueue queue = runsAvail[pageIdxFloor];
-        queue.offer(handle);
+        IntPriorityQueue queue = runsAvail[pageIdxFloor];
+        queue.offer((int) (handle >> BITMAP_IDX_BIT_LENGTH));
```

Comment:
Since `bitmapIdx()` is only used when the handle is a sub-page, I think it'd be prudent to have an `assert isRun(handle)` before we offer the handle to the queue. Our ability to drop the bitmap index information from the handle relies on that assertion holding.